.github/workflows/build-and-test.yml

```
@@ -0,0 +1,60 @@
+    - name: Security check with bandit
+      run: bandit -r src/
+
+    - name: Run tests with pytest
+      run: |
+        pytest --cov=src/ --cov-report=xml
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v4
+      with:
+        file: ./coverage.xml
+        fail_ci_if_error: true
```

yamllint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)